### ISSUE TITLE
Autocompleting columns of `tbl_sql` not supported in cherry blossom

### ISSUE THREAD

**Issue by yuhenghuang** [rstudio/rstudio#12577]
### System details

    RStudio Edition : Server
    RStudio Version : RStudio 2023.01.999-dev+999 "Elsbeth Geranium" Preview (c75a49876b773262c2cc95835652963aea3e56ea, 2023-01-13) for Linux
    OS Version      : Ubuntu 22.04
    R Version       : 4.2.2 Patched (2022-11-10 r83330) -- "Innocent and Trusting"

### Steps to reproduce the problem

```r
library(dplyr)
library(dbplyr)

# prepare DB and tbl_sql
con <- DBI::dbConnect(RSQLite::SQLite(), ":memory:")
dplyr::copy_to(con, mtcars, "mtcars")
tbl <- dplyr::tbl(con, "mtcars")

# autocomplete of column names for tbl_sql doesn't work inside `dplyr::mutate()`
tbl %>%
  mutate()
```

### Describe the problem in detail

I was testing some features of the upcoming version of RStudio [cherry-blossom](https://github.com/rstudio/rstudio/tree/cherry-blossom) when I found that the autocomplete of columns for `tbl_sql` (lazy table from remote DB source) doesn't work in the version.

This feature has been there for years as far as I know and is still being supported in the current version.

After the investigation, I found the [this line](https://github.com/rstudio/rstudio/blob/cherry-blossom/src/cpp/session/modules/SessionRCompletions.R#L114)

is the cause of the problem.

```r
   else if (inherits(object, "data.frame"))
      .rs.acCompletionTypes$DATAFRAME
```

And because remote tables (`tbl_sql`) only inherits `tbl` at base, they are not treated as `DATAFRAME` by the logic.

There are several similar lines in the source file and after fixing those lines I was able to solve the problem on my machine.

The screenshot of fixed behavior follows
![image](https://user-images.githubusercontent.com/41667928/212584848-70450dc0-bdf8-4bd0-8981-b14cbe3b2c56.png)

### Describe the behavior you expected

Since this has been an old feature from the past, I think remote lazy tables should trigger autocompleting of columns in pipe.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by yuhenghuang** [comment]
update:

Modifying the definition of `DATAFRAME` in [source file] (https://github.com/rstudio/rstudio/blob/cherry-blossom/src/cpp/session/modules/SessionRCompletions.R) will force RStudio to retrive the first several rows of the table from DB, which shall not be favorable.

Thus, changing current logic where `tbl_sql` doesn't have column should be sufficient.

The following lines,
[`else if (inherits(object, "data.frame")) type <- .rs.acCompletionTypes$COLUMN`](https://github.com/rstudio/rstudio/blob/cherry-blossom/src/cpp/session/modules/SessionRCompletions.R#L1647)
[`if (inherits(object, "data.frame")) TRUE`](https://github.com/rstudio/rstudio/blob/cherry-blossom/src/cpp/session/modules/SessionRCompletions.R#L2934)
, are what I found to define whether the object owns columns in cherry-blossom.

**Comment by ronblum** [comment]
@yuhenghuang Thank you for raising the issue, as well as for investigating the code! I'll mark this issue for review.

**Comment by ronblum** [comment]
@yuhenghuang Thank you for raising this! I can reproduce this in RStudio Desktop RStudio-2023.03.0-daily+258 on MacOS 13.2. I'll mark this for review as a regression.

Triage: Regression in Cherry Blossom.

**Comment by nilescbn** [comment]
I just upgrated to RStudio 2023.03.0+386 and am experiencing this issue with autocomplete and an Oracle database, which I connect to with the ROracle package and query with dplyr and dbplyr. 

One thing I thought I'd point out is that that autocomplete works within group_by(). I does not work with the other dplyr functions where it used to.

**Comment by Alternikaner** [comment]
Can confirm that the issue persists on 2023.06.01 Build 524

Obviously only a subset of RStudio users regularly interact with DBs, and I would imagine only a sliver of those are faced with tables spanning several hundred columns, but for those of us in this position, column-autocomplete is a key feature. 

Holding thumbs for a bump to next-release milestone status.

**Comment by ronblum** [comment]
Verified fixed in RStudio Desktop 2023.09.0-daily+348 on MacOS 14. Also the ellipses are there thanks to #13196, I believe.


![Image](https://github.com/rstudio/rstudio/assets/31009952/74545b88-4221-4e34-bff1-21bcbe2eac05)
